web-client/public/js/graph.js

```diff
@@ -162,42 +164,63 @@ var drawGraph = function (nodes, links, positiveWeights, negativeWeights, sheetT
       height = newHeight;
     }
 
-    d3.select("svg").attr("width", width).attr("height", height);
+    d3.select("svg").attr("width", newWidth).attr("height", newHeight);
     d3.select("rect").attr("width", width).attr("height", height);
+    d3.select(".boundingBox").attr("width", width).attr("height", height);
     force.size([width, height]).resume();
   });
 
   d3.selectAll("input[name=viewport]").on("change", function () {
-    let value = $(this).attr("value");
-    if (!adaptive && value === "viewportAdapt") {
+    let fixed = $(this).prop("checked");
+    if (!adaptive && !fixed) {
       adaptive = true;
       MAX_SCALE = 10;
       d3.select("rect").attr("stroke", "none");
       zoom.scaleExtent([MIN_SCALE, MAX_SCALE])
       d3.select(".zoomSlider").attr("max", MAX_SCALE);
-    } else if (adaptive && value === "viewportHard") {
-      var BORDER_OFFSET = 4;
-      var newWidth = d3.select(".grnsight-container").style("width");
-      var newHeight = d3.select(".grnsight-container").style("height");
+    } else if (adaptive && fixed) {
+      var newWidth = $(".grnsight-container").css("width");
+      var newHeight = $(".grnsight-container").css("height");
+      width = newWidth.substring(0, newWidth.length - 2) - BORDER_OFFSET;
+      height = newHeight.substring(0, newHeight.length - 2) - BORDER_OFFSET;
 
       adaptive = false;
       MAX_SCALE = 1;
-      d3.select("rect").attr("stroke", "#9A9A9A");
+      d3.select("rect").attr("stroke", "#9A9A9A")
+                       .attr("width", width)
+                       .attr("height", height);
```

Comment:
Will these three lines fit on one at fewer than 120 characters across? Sticks out when compared to the other `d3.select` statements.